src/Data/Storage/Storage.php

```diff
@@ -188,10 +188,10 @@ public function contents($data = null): mixed
     public function deactivate(): IObj {
         // Implement deactivation logic
         $this->perform( State::DISCONNECTING );
-        if ( method_exists($this, '_disconnect') ) {
-        	$this->_disconnect();
+        if ( method_exists($this, 'disconnect') ) {
+        	$this->disconnect();

```

Comment:
This method needs to remain underscored or children can't hook into this injection point